services/QuillLMS/client/app/bundles/Teacher/components/accounts/edit/unlink_modal.jsx

```diff
@@ -24,29 +30,44 @@ export default class UnlinkModal extends React.Component {
   };
 
   handleSubmit = e => {
-    const { updateUser, googleOrClever, } = this.props
+    const { updateUser, provider } = this.props
     const { email, password, } = this.state
     e.preventDefault()
     const data = {
       email,
       password,
       school_options_do_not_apply: true
     };
-    if (googleOrClever === 'Google') {
-      data.google_id = null
-      data.signed_up_with_google = false
-    } else {
-      data.clever_id = null
-    }
-    updateUser(data, '/teachers/update_my_account', `${googleOrClever} unlinked`)
+
+    const providerConfigs = {
+      [PROVIDERS.GOOGLE]: { google_id: null, signed_up_with_google: false },
+      [PROVIDERS.CLEVER]: { clever_id: null },
+      [PROVIDERS.CLASSLINK]: { unlink_classlink: 'true' }
```

Comment:
Is there a reason this is a string and not a boolean?

Reply:
Not a good one ;-) updating